src/rules/color-no-named/index.js

```diff
@@ -0,0 +1,41 @@
+import valueParser from "postcss-value-parser"
+import isCssColorName from "is-css-color-name"
+
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "color-no-named"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: c => `Unexpected named color "${c}"`,
+})
+
+export default function (actual) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { value } = decl
+
+      valueParser(value).walk(function (node) {
+        const charsBeforeColon = decl.toString().indexOf(":")
```

Comment:
Shall we pull this out into a utility? As it's going to get used a lot.

Perhaps something along the lines of `declarationValueIndexOffset` to match the existing [`mediaQueryParamIndexOffset`](https://github.com/stylelint/stylelint/blob/master/src/utils/mediaQueryParamIndexOffset.js)?